### PR TITLE
[PHP81] Support  conditional assignment in ReadOnlyPropertyRector

### LINKED ISSUE #7857: Incorrect behavior of ReadOnlyPropertyRector with conditional assignment in constructor
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version: [0f0b93](https://github.com/rectorphp/rector-src/commit/0f0b93) - released at 2023-03-27 15:06            |

## Minimal PHP Code Causing Issue

https://getrector.com/demo/674bc158-9f21-481b-9c78-4ed12915a9f8

```php
final class DemoFile
{
    private string $env;
    
    public function __construct(bool $sandbox)
    {
        if ($sandbox) {
            $this->env = 'sandbox';
        } else {
            $this->env = 'prod';
        }
    }
}
```

## Expected Behaviour

The `$env` property should be marked as readonly

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/rectorphp/rector/issues/7857

### CODE REVIEW

File: rules/TypeDeclaration/AlreadyAssignDetector/ConstructorAssignDetector.php [hunk 1]
```diff
@@ -106,6 +109,70 @@ private function decorateFirstLevelStatementAttribute(array $classMethods): void
...
+                            $assign = $ifStmt->expr;
+                        }
+
+                        if ($assign instanceof Assign && $assign->var instanceof Expr\PropertyFetch) {
+                            $name = $this->nodeNameResolver->getName($assign->var);
+                            $alwaysAssignedProperties[$name] = $assign;
+                        }
+                    }
+
+                    /*
```
Comment (staabm) [inline rules/TypeDeclaration/AlreadyAssignDetector/ConstructorAssignDetector.php:128]: the PR as is works for IF/ELSE constructions, but not yet for IF/ELSEIF/ELSE constructions.

I can do this last piece after we agreed on the approach.

any feedback? should I proceed?

Comment (samsonasik) [reply rules/TypeDeclaration/AlreadyAssignDetector/ConstructorAssignDetector.php:128]: I am curious if there is already PHPStan attribute for "always assigned, like `assigned_to`, ref https://github.com/search?q=repo%3Arectorphp%2Frector-src%20ASSIGNED_TO&type=code

**Comment by TomasVotruba** [comment]
What needs to be done here to merge? Or shall we close and look for another way?

**Comment by staabm** [comment]
I have no interesst to finish it.

**Pull request closed without merging**